src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/claimsbyhearingvenue/ClaimsByHearingVenueReport.java

```diff
@@ -1,45 +1,34 @@
 package uk.gov.hmcts.ethos.replacement.docmosis.reports.claimsbyhearingvenue;
 
-import lombok.RequiredArgsConstructor;
 import org.apache.commons.collections4.CollectionUtils;
 import org.apache.commons.lang3.StringUtils;
-import org.springframework.stereotype.Service;
 import uk.gov.hmcts.ecm.common.helpers.UtilHelper;
 import uk.gov.hmcts.ecm.common.model.ccd.items.RespondentSumTypeItem;
 import uk.gov.hmcts.ecm.common.model.ccd.types.ClaimantType;
 import uk.gov.hmcts.ecm.common.model.ccd.types.ClaimantWorkAddressType;
 import uk.gov.hmcts.ecm.common.model.reports.claimsbyhearingvenue.ClaimsByHearingVenueSubmitEvent;
 import uk.gov.hmcts.ethos.replacement.docmosis.helpers.ReportHelper;
-import uk.gov.hmcts.ethos.replacement.docmosis.reports.ReportParams;
 
 import java.time.LocalDate;
 import java.util.Comparator;
 import java.util.List;
 import static uk.gov.hmcts.ecm.common.model.helper.Constants.CLAIMS_BY_HEARING_VENUE_REPORT;
 import static uk.gov.hmcts.ecm.common.model.helper.Constants.SINGLE_HEARING_DATE_TYPE;
 
-@RequiredArgsConstructor
-@Service
 public class ClaimsByHearingVenueReport {
     private static final String NULL_STRING_VALUE = "Null";
-    private ClaimsByHearingVenueReportDataSource reportDataSource;
-    private ReportParams searchParams;
 
-    public ClaimsByHearingVenueReport(ClaimsByHearingVenueReportDataSource dataSource, ReportParams params) {
-        reportDataSource = dataSource;
-        searchParams = params;
-    }
-
-    public ClaimsByHearingVenueReportData generateReport(String hearingDateType, String userFullName) {
-        var submitEvents = reportDataSource.getData(
-            UtilHelper.getListingCaseTypeId(searchParams.getCaseTypeId()),
-                searchParams.getDateFrom(), searchParams.getDateTo());
-        var claimsByHearingVenueReportData = initReport(searchParams.getCaseTypeId());
-
-        setReportListingDate(claimsByHearingVenueReportData, searchParams.getDateFrom(),
-                searchParams.getDateTo(), hearingDateType);
-
-        claimsByHearingVenueReportData.setReportPrintedOnDescription(getReportedOnDetail(userFullName));
+    public ClaimsByHearingVenueReportData generateReport(ClaimsByHearingVenueReportDataSource dataSource,
```

Comment:
dataSource is a dependency not a parameter. There is no difference between dataSource and a repository or a service

Reply:
fixed.